allennlp/data/dataset_readers/coreference_resolution/util.py

```diff
@@ -47,6 +49,31 @@ def make_coref_instance(
                 how many spans we are considering), we represent this a as a `SequenceLabelField`
                 with respect to the `spans `ListField`.
     """
+    if max_sentences is not None and len(sentences) > max_sentences:
```

Comment:
Ah! This makes the `max_sentences` description a little off, because really this is randomly selecting a `max_sentences` size chunk from the document. Could you rename the parameter so it is more appropriate? something like `sample_to_length` or something.

Reply:
Why do you think the `max_sentences` description is a little off? Because we are sampling instead of truncating? If so what about `max_sentences_to_sample`? I think `sample_to_length` is a little underspecified as to whether it's the number of sentences or tokens.

Reply:
yeah, `max_sentences_to_sample` is better 👍  Basically just that if something says `max_sentences`, I expect it to be strictly a limit, and not an indication that some sampling is happening.